### PR TITLE
Load metadata for tables without table rule

### LINKED ISSUE #10205: The metadata of the tables without sharding rule is not refreshed after ddl executed

### PR DESCRIPTION
**Pull request by menghaoranss**
Fixes #10205 .

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java [hunk 1]
```diff
@@ -38,7 +39,8 @@ public void refresh(final ShardingSphereSchema schema, final Collection<String>
                         final AlterTableStatement sqlStatement, final SchemaBuilderMaterials materials) throws SQLException {
         String tableName = sqlStatement.getTable().getTableName().getIdentifier().getValue();
         if (!containsInTableContainedRule(tableName, materials)) {
-            return;
+            TableMetaDataLoader.load(materials.getDataSourceMap().get(routeDataSourceNames.iterator().next()), 
+                    tableName, materials.getDatabaseType()).ifPresent(tableMetaData -> schema.put(tableName, tableMetaData));
         }
         if (null != schema && schema.containsTable(tableName)) {
             TableMetaDataBuilder.build(tableName, materials).ifPresent(tableMetaData -> schema.put(tableName, tableMetaData));
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java:46]: Do you think `if() {} else if() {}`  is more appropriate?

Comment (tristaZero) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java:46]: Beside, we still need to give a check on `routeDataSourceNames.size()`

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java [hunk 2]
```diff
@@ -44,7 +45,7 @@ public void refresh(final ShardingSphereSchema schema, final Collection<String>
         if (containsInTableContainedRule(tableName, materials)) {
             tableMetaData = TableMetaDataBuilder.build(tableName, materials).orElse(new TableMetaData());
         } else {
-            tableMetaData = new TableMetaData();
+            tableMetaData = TableMetaDataLoader.load(materials.getDataSourceMap().get(routeDataSourceNames.iterator().next()), tableName, materials.getDatabaseType()).orElse(new TableMetaData());
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java:48]: Is  it necessary to check `routeDataSourceNames.size()`

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresherTest.java [hunk 3]
```diff
@@ -31,90 +36,79 @@
...
-    
-    @Test
-    public void refreshWithUnConfiguredForSQL92() throws SQLException {
-        SQL92CreateTableStatement createTableStatement = new SQL92CreateTableStatement();
-        refreshWithUnConfigured(createTableStatement);
-    }
-    
-    @Test
-    public void refreshWithUnConfiguredForSQLServer() throws SQLException {
-        SQLServerCreateTableStatement createTableStatement = new SQLServerCreateTableStatement();
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresherTest.java:111]: Do we add a UT to cover the branch of `!containsInTableContainedRule(tableName, materials)` ?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java (hunk 2 above)
Comment (menghaoranss) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java:48]: The tables in DDL and not in table rule,  there should be only one route data source，may be we should check `routeDataSourceNames.size() !=0` ?

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresherTest.java (hunk 3 above)
Comment (menghaoranss) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresherTest.java:111]: Added.

**Review by tristaZero — APPROVED** [review]

**Review by tristaZero — CHANGES_REQUESTED** [review]
CI failed.
Plus,

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java [hunk 4]
```diff
@@ -38,9 +42,12 @@ public void refresh(final ShardingSphereSchema schema, final Collection<String>
                         final AlterTableStatement sqlStatement, final SchemaBuilderMaterials materials) throws SQLException {
         String tableName = sqlStatement.getTable().getTableName().getIdentifier().getValue();
         if (!containsInTableContainedRule(tableName, materials)) {
-            return;
-        }
-        if (null != schema && schema.containsTable(tableName)) {
+            DataSource dataSource = materials.getDataSourceMap().get(routeDataSourceNames.iterator().next());
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java:45]: Is there any change `null == dataSource`?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java [hunk 5]
```diff
@@ -38,9 +42,12 @@ public void refresh(final ShardingSphereSchema schema, final Collection<String>
...
+            TableMetaData tableMetaData = Objects.isNull(dataSource) ? new TableMetaData()
+                    : TableMetaDataLoader.load(materials.getDataSourceMap().get(routeDataSourceNames.iterator().next()),
+                    tableName, materials.getDatabaseType()).orElse(new TableMetaData());
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/AlterTableStatementSchemaRefresher.java:48]: if the first ds has no this table, how about the remaining ds?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java [hunk 6]
```diff
@@ -58,4 +62,18 @@ private boolean containsInTableContainedRule(final String tableName, final Schem
         }
         return false;
     }
+    
+    private TableMetaData loadTableMetaData(final String tableName, final Collection<String> routeDataSourceNames, 
+                                            final SchemaBuilderMaterials materials) throws SQLException {
+        for (String routeDataSourceName : routeDataSourceNames) {
+            DataSource dataSource = materials.getDataSourceMap().get(routeDataSourceName);
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java:69]: Is it possible that `Objects.isNull(dataSource)` == true?

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Review by tristaZero — APPROVED** [review]

**Pull request merged**